MdePkg/Library/BaseRngLib/Rand/RdRand.c

```diff
@@ -48,10 +131,13 @@ BaseRngLibConstructor (
   // CPUID. A value of 1 indicates that processor support RDRAND instruction.
   //
   AsmCpuid (1, 0, 0, &RegEcx, 0);
-  ASSERT ((RegEcx & RDRAND_MASK) == RDRAND_MASK);
 
   mRdRandSupported = ((RegEcx & RDRAND_MASK) == RDRAND_MASK);
 
+  if (mRdRandSupported) {
+    mRdRandSupported = TestRdRand ();
+  }

```

Comment:
Missing RDRAND support should be considered fatal under a confidential VM too, as there's no other trustworthy source of random bytes.

Reply:
No disagreement here, but this is arguably orthogonal - this particular driver should simply not pretend that RDRAND is available if it is not (and crash or give bogus outputs). Whether a policy is in effect that requires this to be a fatal error is up to the platform to decide, and therefore does not belong in the core RDRAND RngLib

Reply:
Sure, I don't know enough about EDK2 to suggest where in the code is best to perform such a check, just wanted to raise it as an issue to consider, since IIUC this code is being included in the SSEV/TDX  targeted builds